src/test/backup.test.ts

```diff
@@ -0,0 +1,21 @@
+/*---------------------------------------------------------
+ * Copyright (C) Microsoft Corporation. All rights reserved.
+ *--------------------------------------------------------*/
+
+import { expect } from "chai";
+import { HexDocumentEdit, HexDocumentEditOp } from "../../shared/hexDocumentModel";
+import { Backup } from "../backup";
+import { getTempFile } from "./util";
+
+describe("Backup", () => {
+	const edits: HexDocumentEdit[] = [
+		{ op: HexDocumentEditOp.Delete, offset: 3, opId: 0, previous: new Uint8Array([3, 4, 5]) },
+		{ op: HexDocumentEditOp.Replace, offset: 1, opId: 0, value: new Uint8Array([10, 11, 12]), previous: new Uint8Array([1, 2, 6]) },
+	];
+
+	it("round trips", async () => {
+		const backup = new Backup(await getTempFile());
+		await backup.write(edits);
+		expect(await backup.read()).to.deep.equal(edits);
```

Comment:
Is there a benefit to chai over mocha's built in `assert.strictDeepEqual`?

Reply:
For `strictDeepEqual`, not so much, but Chai has additional assertions which makes writing tests easier and/or produces higher quality assertion messages, as well as a plugin API. I don't feel super strongly about it though so I'm fine using just `assert` if you'd prefer.

Reply:
If you think chai will be beneficial in the future let's keep it. However, do you think we should be using Jest and React's testing library https://testing-library.com/docs/react-testing-library/intro/ so we can test component rendering and those sorts of things?